tests/ui/pattern/usefulness/integer-ranges/gap_between_ranges.stderr

```diff
@@ -0,0 +1,138 @@
+error: multiple ranges are one apart
+  --> $DIR/gap_between_ranges.rs:16:9
+   |
+LL |         20..30 => {}
+   |         ^^^^^^
+   |         |
+   |         this range doesn't match `30_u8` because `..` is a non-inclusive range
+   |         help: use an inclusive range instead: `20_u8..=30_u8`
+LL |         31..=40 => {}
+   |         ------- this seems to continue range `20_u8..30_u8`, but `30_u8` isn't matched by either of them
+   |
+note: the lint level is defined here
+  --> $DIR/gap_between_ranges.rs:2:9
+   |
+LL | #![deny(small_gaps_between_ranges)]
+   |         ^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error: multiple ranges are one apart
+  --> $DIR/gap_between_ranges.rs:21:9
+   |
+LL |         20..30 => {}
+   |         ^^^^^^
+   |         |
+   |         this range doesn't match `30_u8` because `..` is a non-inclusive range
+   |         help: use an inclusive range instead: `20_u8..=30_u8`
+LL |         31 => {}
+   |         -- this seems to continue range `20_u8..30_u8`, but `30_u8` isn't matched by either of them
+
+error: multiple ranges are one apart
+  --> $DIR/gap_between_ranges.rs:26:13
+   |
+LL |     m!(0u8, 20..30, 31..=40);
+   |             ^^^^^^  ------- this seems to continue range `20_u8..30_u8`, but `30_u8` isn't matched by either of them
+   |             |
+   |             this range doesn't match `30_u8` because `..` is a non-inclusive range
+   |             help: use an inclusive range instead: `20_u8..=30_u8`
+
+error: multiple ranges are one apart
+  --> $DIR/gap_between_ranges.rs:27:22
+   |
+LL |     m!(0u8, 31..=40, 20..30);
+   |             -------  ^^^^^^
+   |             |        |
+   |             |        this range doesn't match `30_u8` because `..` is a non-inclusive range
+   |             |        help: use an inclusive range instead: `20_u8..=30_u8`
+   |             this seems to continue range `20_u8..30_u8`, but `30_u8` isn't matched by either of them
+
+warning: multiple patterns overlap on their endpoints
+  --> $DIR/gap_between_ranges.rs:28:21
+   |
+LL |     m!(0u8, 20..30, 29..=40);
+   |             ------  ^^^^^^^ ... with this range
+   |             |
+   |             this range overlaps on `29_u8`...
+   |
+   = note: you likely meant to write mutually exclusive ranges
+   = note: `#[warn(overlapping_range_endpoints)]` on by default
```

Comment:
We should probably update this lint to match the style of the new one.

Reply:
Hm they could even be part of a same lint if we can find a good name

Reply:
@Nadrieril sadly we can't really rename `overlapping_range_endpoints` in practice, as it already existed :-/

Reply:
Oh yeah :/ We better find a good name for this one then. `non_contiguous_range_endpoints`?

Reply:
By that name it would be consistent to lint on:
```rust
match x {
   0..=10 => {}
   12..=20 => {} // WARN: `11` is skipped over
   _ => {}
}
```
That feels unlikely to be a mistake though, would you say it's ok to only lint for exclusive ranges despite the name?

Reply:
I think that name is still reasonable even though it only apply to one of the two types of ranges.